warehouse/query-core/src/test/java/datawave/query/jexl/functions/EvaluationPhaseFilterFunctionsTest.java

```diff
@@ -476,46 +476,97 @@ private boolean resultForDefaultOperator(int count) {
         // Verify that a non-null value tuple returns a set with that value tuple.
         @Test
         public void testNonNullValue() {
             ValueTuple valueTuple = toValueTuple("FOO.1,BAR,bar");
-            givenFieldValue(valueTuple);
+            givenFieldValue1(valueTuple);
             assertThat(result()).containsExactly(valueTuple);
         }
         
+        // Verify that a non-null value tuple returns a set with that value tuple with use of multiple filter fields.
+        @Test
+        public void testNonNullValuesWithMultipleFields() {
+            ValueTuple valueTuple = toValueTuple("F00.1,BAR,bar");
+            givenFieldValue1(valueTuple);
+            givenFieldValue2("F00.1");
+            
+            assertThat(resultForMultipleFieldValues()).containsExactly(valueTuple);
+        }
+        
         // Verify that an empty collection returns an empty set.
         @Test
         public void testEmptyCollection() {
-            givenFieldValue(Collections.emptySet());
+            givenFieldValue1(Collections.emptySet());
             assertThat(result()).isEmpty();
         }
         
+        // Verify that an empty collection returns an empty set with use of multiple filter fields.
+        @Test
+        public void testEmptyCollectionWithMultipleFields() {
+            givenFieldValue1("FOO.1");
+            givenFieldValue2(Collections.emptySet());
+            
+            assertThat(resultForMultipleFieldValues()).isEmpty();
+        }
+        
         // Verify that a non-empty collection of value tuples returns a set with the tuples.
         @Test
         public void givenNonEmptyCollection() {
             ValueTuple valueTuple1 = toValueTuple("FOO.1,BAR,bar");
             ValueTuple valueTuple2 = toValueTuple("FOO.1,ZOOM,zoom");
-            givenFieldValue(Sets.newHashSet(valueTuple1, valueTuple2));
+            givenFieldValue1(Sets.newHashSet(valueTuple1, valueTuple2));
             assertThat(result()).containsExactlyInAnyOrder(valueTuple1, valueTuple2);
         }
         
-        private void givenFieldValue(Object fieldValue) {
-            this.fieldValue = fieldValue;
+        // Verify that a non-empty collection of value tuples returns a set of the provided tuples,
+        // while utilizing multiple filter fields.
+        @Test
+        public void givenNonEmptyCollectionWithMultipleFields() {
+            ValueTuple valueTuple1 = toValueTuple("FOO.1,BAR,bar");
+            ValueTuple valueTuple2 = toValueTuple("FOO.1,ZOOM,zoom");
+            givenFieldValue1(Sets.newHashSet(valueTuple1, valueTuple2));
+            
+            ValueTuple valueTuple3 = toValueTuple("FOO.2,BAT,bat");
+            ValueTuple valueTuple4 = toValueTuple("FOO.2,ZAP,zap");
+            givenFieldValue2(Sets.newHashSet(valueTuple3, valueTuple4));
+            
+            assertThat(resultForMultipleFieldValues()).containsExactlyInAnyOrder(valueTuple1, valueTuple2);
+        }
+        
```

Comment:
I'm reviewing the tests before the code and am thinking it might be worth adding more tests.  Here are a few scenarios to consider:
1. givenFieldValue1 is a collection containing multiple entries and givenFieldValue2 is empty
2. reverse of scenario 1, where value1 is empty and value2 is a collection
3. givenFieldValue1 is a collection containing multiple entries and givenFieldValue2 is null
4. reverse of scenario 3
5. reverse of testEmptyCollectionWithMultipleFields, where value2 is a non-empty collection and value1 is empty
6. reverse of testNullValueWithMultipleFields with null for 2, value 1 is a non-empty string
7. both value1 and value 2 are null
8. both value1 and value 2 are empty

Reply:
There have been significant changes from the initial MR, including the unit tests.  Can you please re-review when you have a chance.  Thanks!